src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/GenericWhitespaceCheck.java

```diff
@@ -261,9 +260,8 @@ private void processStart(DetailAST ast) {
             // Detect if the first case
             final DetailAST parent = ast.getParent();
             final DetailAST grandparent = parent.getParent();
-            if (parent.getType() == TokenTypes.TYPE_PARAMETERS
```

Comment:
If we know we are on method/def and current token is generic, then the token inbetween cant be anything other then type parameters.